WalletWasabi/Wallets/Wallet.cs

```diff
@@ -471,6 +473,14 @@ private async Task LoadWalletStateAsync(CancellationToken cancel)
 
 		Height bestKeyManagerHeight = KeyManager.GetBestTurboSyncHeight();
 
+		// Make sure best height is at least height of segwit activation.
+		var startingSegwitHeight = new Height(SmartHeader.GetStartingHeader(Network, IndexType.SegwitTaproot).Height);
+		if (startingSegwitHeight > bestKeyManagerHeight)
```

Comment:
So if `TurboSyncHeight` is lower then you will set both `Height` to activation. It sounds OK. What if an old wallet without Turbo opened with this version? Will it set back the Height to the activation, generating a rescan?

Reply:
No, At this point `Height` is necessarily `>=` or lower than `TurboSyncHeight`

Reply:
So if `TurboSyncHeight` is set higher by this, `Height` will also be set higher. It's impossible that `Height` will be set lower because of this.

Reply:
BTW the comment I linked is wrong, `EnsureTurboSyncHeightConsistency` makes sure that it's not **lower** 
https://github.com/zkSNACKs/WalletWasabi/blob/a4347fa975380d2e9891a6fde34e446607337281/WalletWasabi/Blockchain/Keys/KeyManager.cs#L743-L747C5

Reply:
Can you check and set them individually? What I don't like here is the logic of if Turbo is broken somehow it will touch the Height.

Reply:
If this assumption is incorrect, the wallet here is in a completely broken state. All the other parts of the synchronization will fall apart, and the user might very well discover that he's Satoshi. Anyway I will change it, which will make the code ugly, twice as big, less performant, and inconsistent with the rest of the software.